internal/system/agent/executor/docker_exec.go

```diff
@@ -0,0 +1,134 @@
+	internal.SupportLoggingServiceKey:       "edgex-support-logging",
+	internal.ConfigSeedServiceKey:           "edgex-config-seed",
+}
+
+func FindDockerContainerStatus(service string, status string) bool {
+
+	var (
+		cmdOut []byte
+		err    error
+	)
+	cmdName := "docker"
+	cmdArgs := []string{"ps"}
+	if cmdOut, err = exec.Command(cmdName, cmdArgs...).Output(); err != nil {
+		logs.LoggingClient.Error("error running the docker command", "error message", err.Error())
+		os.Exit(1)
+	}
+	dockerOutput := string(cmdOut)
+	// Find whether the container to start has started.
+	for _, line := range strings.Split(strings.TrimSuffix(dockerOutput, "\n"), "\n") {
+		if strings.Contains(line, service) {
+
+			if status == "Up" {
+				if strings.Contains(line, "Up") {
+					logs.LoggingClient.Debug("container started", "service name", service, "details", line)
+					return true
+				} else {
+					logs.LoggingClient.Warn("container NOT started", "service name", service)
+					return false
+				}
+			} else if status == "Exited" {
+				if strings.Contains(line, "Exited") {
+					logs.LoggingClient.Debug("container stopped", "service name", service, "details", line)
+					return true
+				} else {
+					logs.LoggingClient.Warn("container NOT stopped", "service name", service)
+					return false
+				}
+			}
+		}
+	}
+	return false
+}
+
+func ExecuteDockerCommands(service string, operation string) error {
+	_, knownService := services[service]
+
+	if knownService {
+		RunDockerCommands(service, services[service], operation)
+
+		return nil
+	} else {
+		newError := fmt.Errorf("unknown service: %v", service)
+		logs.LoggingClient.Error(newError.Error())
+
+		return newError
+	}
+}
+
+func RunDockerCommands(service string, dockerService string, operation string) {
+
+	var (
+		err    error
+		cmdDir string
+	)
+
+	cmdName := "docker"
+
+	// Retry finding the path to where the docker command will be run.
+	err = Do(func(attempt int) (bool, error) {
+		var err error
+		cmdDir, err = FindPathToRunDocker()
+		// Try 5 times
+		return attempt < 5, err
+	})
+	if err != nil {
+		logs.LoggingClient.Error("unable to find the path to where the docker command will be run", "error message", err.Error())
+	}
+
+	cmdArgs := []string{operation, dockerService}
+	cmd := exec.Command(cmdName, cmdArgs...)
+	cmd.Dir = cmdDir
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		logs.LoggingClient.Warn("docker command failed", "error message", err.Error())
+		logs.LoggingClient.Warn("associated ouptut", "error message", out)
+	}
+
+	if operation == "start" {
+		if ! FindDockerContainerStatus(service, "Up") {
+			logs.LoggingClient.Warn("docker start operation failed", "service name", service)
+		}
+	} else if operation == "stop" {
+		if ! FindDockerContainerStatus(service, "Exited") {
+			logs.LoggingClient.Warn("docker stop operation failed", "service name", service)
+		}
+	}
+}
+
+func FindPathToRunDocker() (string, error) {
```

Comment:
Function isn't called outside of this package, so doesn't need to be exported (capitalized)